src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -490,3 +526,201 @@ def execute_hdi_script_action(cmd, client, resource_group_name, cluster_name,
     ]
 
     return client.execute_script_actions(resource_group_name, cluster_name, persist_on_success, script_actions_params)
+
+
+# pylint: disable=redefined-builtin
+def create_autoscale(cmd, client, resource_group_name, cluster_name, type, min_count=None,
+                     max_count=None, timezone=None, days=None, time=None, count=None):
+    from azure.mgmt.hdinsight.models import Autoscale, AutoscaleCapacity, AutoscaleRecurrence, AutoscaleSchedule, \
+        AutoscaleTimeAndCapacity
+    autoscale_configuration = None
+    if type == 'Load':
+        if not all([min_count, max_count]):
+            raise CLIError(
+                'When the autoscale type is Load, both the min_count and max_count should be specified.')
+
+        autoscale_configuration = Autoscale(
+            capacity=AutoscaleCapacity(
+                min_instance_count=min_count,
+                max_instance_count=max_count
+            )
+        )
+    elif type == 'Schedule':
+        if not all([timezone, days, time, count]):
+            raise CLIError(
+                'When the autoscale type is Schedule, all of the timezone, days, time, count should be specified.')
+
+        autoscale_configuration = Autoscale(
+            recurrence=AutoscaleRecurrence(
+                time_zone=timezone,
+                schedule=[AutoscaleSchedule(
+                    days=days,
+                    time_and_capacity=AutoscaleTimeAndCapacity(
+                        time=time,
+                        min_instance_count=count,
+                        max_instance_count=count
+                    )
+                )]
+            )
+        )
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def update_autoscale(cmd, client, resource_group_name, cluster_name, min_count=None, max_count=None, timezone=None):
```

Comment:
>update_autoscale [](start = 4, length = 16)

what will happen if `--min-count` `--max-count` and `--timezone` are specified at the same time?

Reply:
No matter the cluster is Load-based or Schedule-based, it will fail. Because --min-count is Load-based's parameter, and --timezone is Schedule-based's parameter, if all are specified, Rp will return error "both recurrence and capacity cannot be defined with autoscale."

---
In reply to: [466351317](https://github.com/Azure/azure-cli/pull/14692#discussion_r466351317) [](ancestors = 466351317)